pallets/substrate-validator-set/src/lib.rs

```diff
@@ -141,15 +167,171 @@ pub mod pallet {
 			Flag::<T>::put(true);
 			Ok(().into())
 		}
+
+		// Activate Validator takes in a validator request ID
+		// Based on that ID, it can fetch the validator request from storage
+		// check if the signer is mentioned as the council address from the request (this way we know if the signer is an active council member)
+		// if true, then we add the validator account from the request to the list of validators
+		#[pallet::weight(0)]
+		pub fn activate_validator(
+			origin: OriginFor<T>,
+			validator_request_id: u32,
+		) -> DispatchResultWithPostInfo {
+			let address = ensure_signed(origin)?;
+
+			let mut validator_request = <ValidatorRequest<T>>::get(validator_request_id)
+				.ok_or(DispatchError::from(Error::<T>::ValidatorRequestNotFound))?;
+
+			ensure!(
+				validator_request.state == types::ValidatorRequestState::Approved,
+				Error::<T>::ValidatorRequestNotApproved
+			);
+			ensure!(
+				validator_request.council_account == address,
+				Error::<T>::UnauthorizedToActivateValidator
+			);
+
+			// Update the validator request
+			validator_request.state = types::ValidatorRequestState::Executed;
+			<ValidatorRequest<T>>::insert(validator_request_id, &validator_request);
+
+			Self::do_add(validator_request.validator_account);
+			// Calling rotate_session to queue the new session keys.
+			Session::<T>::rotate_session();
+			// Triggering rotate session again for the queued keys to take effect.
+			Flag::<T>::put(true);
+
+			Ok(().into())
+		}
+
+		// create a request to become a validator
+		// Validator account: the account of the validator
+		// Stash account: the "bank" account of the validator (where rewards should be sent to)
+		// Description: why someone wants to become a validator
+		// Tf Connect ID: the threefold connect ID of the persion who wants to become a validator
+		// Info: some public info about the validator (website link, blog link, ..)
+		// A user can only have 1 validator request at a time
+		#[pallet::weight(0)]
```

Comment:
why is this weight set to 0?

Reply:
Good question, although not sure what weight we should set. In all our pallets we use different weights, I think it's time we review those as well.